paddle/fluid/inference/api/demo_ci/inference_icnet_preprocess.h

```diff
@@ -0,0 +1,189 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <Windows.h>
+#include <algorithm>
+#include <cassert>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+// note the image pre-process works for icnet only
+const int C = 3;    // image channel
+const int H = 449;  // image height
+const int W = 581;  // image width
+const float Image_Mean[3] = {112.15f, 109.41f, 185.42f};
+
+struct Record {
+  float* data;
+  std::vector<int32_t> shape;
+  Record() { data = new float[C * H * W]; }
+  ~Record() {
+    if (data != nullptr) {
+      delete data;
+      data = nullptr;
+    }
+  }
+};
+
+class ImageProcess {
+ public:
+  static bool image_read(std::string imagename, unsigned char*& imagebuf,
+                         int& imagewidth, int& imageheight,
+                         int& imagebitcount) {
+    int imagelinebyte = 0;
+    FILE* filep;
+    errno_t filerror = fopen_s(&filep, &imagename[0], "rb");
+    if (filerror == 0) {
+      if (!fseek(filep, sizeof(BITMAPFILEHEADER), 0)) {
+        BITMAPINFOHEADER Image_Header;
+        fread(&Image_Header, sizeof(BITMAPINFOHEADER), 1, filep);
+        imagewidth = Image_Header.biWidth;
+        imageheight = Image_Header.biHeight;
+        imagebitcount = Image_Header.biBitCount;
+        imagelinebyte = (imagewidth * imagebitcount / 8 + 3) / 4 * 4;
+        if (imagebitcount == 8) {
+          fread(new RGBQUAD[256], sizeof(RGBQUAD), 256, filep);
+        }
+        imagebuf = new unsigned char[imagelinebyte * imageheight];
+        fread(imagebuf, 1, imagelinebyte * imageheight, filep);
+        unsigned char* imagedatabuf =
+            new unsigned char[imagewidth * imageheight * 3];
+        for (int i = 0; i < imageheight; i++) {
+          for (int j = 0; j < imagewidth; j++) {
+            for (int k = 0; k < 3; k++) {
+              imagedatabuf[k * imagewidth * imageheight + i * imagewidth + j] =
+                  imagebuf[i * imagelinebyte + j * 3 + k];
+            }
+          }
+        }
+        delete imagebuf;
+        imagebuf = imagedatabuf;
+      }
+      fclose(filep);
+
+      return true;
+    }
+
+    return false;
+  }
+
+  // save the data into BMP file
+  static bool image_save(const char* imagename, unsigned char* imagebuf,
+                         int imagewidth, int imageheight, int imagebitcount) {
+    if (!imagebuf) return 0;
+    int colorTablesize = 0;
+    if (imagebitcount == 8) {
+      colorTablesize = 256;
+    }
+    int imagelinebyte = (imagewidth * imagebitcount / 8 + 3) / 4 * 4;
+    FILE* fp;
+    fopen_s(&fp, imagename, "wb");
+    if (fp == 0) {
+      return false;
+    }
+    BITMAPFILEHEADER fileHead;

```

Comment:
not a good design,  put format specific code into a wrapper class.

Reply:
Romove them